### ISSUE TITLE
webpack@5: Error: Module.entryModule: Multiple entry modules are not supported by the deprecated API (Use the new ChunkGroup API)

### ISSUE THREAD

**Issue by casdevs** [xz64/license-webpack-plugin#67]
When using `license-webpack-plugin@2.1.3` with the new `webpack@5.0.0-beta.7`, the following error occurs:

```
Error: Module.entryModule: Multiple entry modules are not supported by the deprecated API (Use the new ChunkGroup API)
```

callstack:
```
Error: Module.entryModule: Multiple entry modules are not supported by the deprecated API (Use the new ChunkGroup API)
    at Chunk.get entryModule [as entryModule] (/home/node/app/node_modules/webpack/lib/Chunk.js:113:10)
    at WebpackChunkModuleIterator.iterateModules (/home/node/app/node_modules/license-webpack-plugin/dist/WebpackChunkModuleIterator.js:38:19)
    at PluginChunkReadHandler.processChunk (/home/node/app/node_modules/license-webpack-plugin/dist/PluginChunkReadHandler.js:26:29)
    at _loop_1 (/home/node/app/node_modules/license-webpack-plugin/dist/WebpackCompilerHandler.js:48:37)
    at WebpackCompilerHandler.iterateChunks (/home/node/app/node_modules/license-webpack-plugin/dist/WebpackCompilerHandler.js:71:17)
    at /home/node/app/node_modules/license-webpack-plugin/dist/WebpackCompilerHandler.js:29:27
    at Hook.eval [as callAsync] (eval at create (/home/node/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:32:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/node/app/node_modules/tapable/lib/Hook.js:18:14)
    at /home/node/app/node_modules/webpack/lib/Compilation.js:1592:37
    at Hook.eval [as callAsync] (eval at create (/home/node/app/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (/home/node/app/node_modules/tapable/lib/Hook.js:18:14)
    at cont (/home/node/app/node_modules/webpack/lib/Compilation.js:1586:33)
    at /home/node/app/node_modules/webpack/lib/Compilation.js:1638:6
    at /home/node/app/node_modules/neo-async/async.js:2830:7
    at Object.each (/home/node/app/node_modules/neo-async/async.js:2850:39)
    at Compilation.createChunkAssets (/home/node/app/node_modules/webpack/lib/Compilation.js:2308:12)
```

my (simplified) webpack config is as follows:
```
{
  entry: {
    index: [
      'core-js/stable',
      'regenerator-runtime/runtime'
      'index.js'
   ],
   mode: 'production',
   target: 'web',
   plugins: [
     new LicenseWebpackPlugin(),
     ...
    ],
   ...
}
``` 

Most of the other plugins we use simply throw deprecation warnings rather than an error.

Do others have similar problems when using webpack@5?
Are there any plans to test / increase compatibility with the upcoming webpack version?

**Comment by xz64** [comment]
I plan to fix the deprecation errors

**Comment by casdevs** [comment]
thanks, great, highly appreciated!

**Comment by xz64** [comment]
Looks like the module information is not available during the OptimizeChunkAssets hook via the ChunkGraph API. I may have to look into rewriting major portions of the plugin for webpack 5.

**Comment by casdevs** [comment]
thanks for the feedback, would be great!
If you need help in testing new code, I'm happy to support you if I can.

**Comment by silverwind** [comment]
Any news on this one? I think once it's fixed, this module should be compatible with webpack 5.

**Comment by xz64** [comment]
I am working on it

**Comment by xz64** [comment]
Fixed in v2.3.3
